src/main/java/com/google/sps/servlets/utils/CommentAnalysis.java

```diff
@@ -53,31 +53,56 @@ public CommentAnalysis(LanguageServiceClient languageService) {
    * @return a Statistics object that contains required values to display
    */
   public Statistics computeOverallStats(CommentThreadListResponse youtubeResponse) {
-    List<Double> scoreValues =
+    // all sentiments built by comment content retrieved from youtubeResponse
+    List<Sentiment> documentList =
         youtubeResponse.getItems().stream()
             .map(UserComment::new)
-            .map(this::calcualateSentiAnalysisScore)
+            .map(
+                comment ->
+                    languageService
+                        .analyzeSentiment(
+                            Document.newBuilder()
+                                .setContent(comment.getCommentMsg())
+                                .setType(Document.Type.PLAIN_TEXT)
+                                .build())
+                        .getDocumentSentiment())
+            .collect(Collectors.toList());
+    // List of scores and magnitude for all comments
+    List<Double> scoreValues =
+        documentList.stream().map(this::calcualateSentiAnalysisScore).collect(Collectors.toList());
+    List<Double> magnitudeValues =
+        documentList.stream()
+            .map(this::calculateSentiAnalysisMagnitude)
             .collect(Collectors.toList());
-    return new Statistics(scoreValues);
+    return new Statistics(scoreValues, magnitudeValues);
   }
 
   /**
    * Perform sentiment analysis of comment.
    *
+   * @param sentiment object retrieved from document for each comment
    * @return sentiment score
    * @throws RuntimeException if the sentiment analysis API is not working, throw the IOExeption
    */
-  private double calcualateSentiAnalysisScore(UserComment comment) {
-    Document doc =
-        Document.newBuilder()
-            .setContent(comment.getCommentMsg())
-            .setType(Document.Type.PLAIN_TEXT)
-            .build();
-    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+  private double calcualateSentiAnalysisScore(Sentiment sentiment) {
     if (sentiment != null) {
       return ((double) sentiment.getScore());
     } else {
-      throw new RuntimeException("Failed to build the sentiments");
+      throw new RuntimeException("Failed to get the sentiment score");
+    }
+  }
+
+  /**
+   * Perform sentiment analysis of comment.
+   *
+   * @return sentiment score
+   * @throws RuntimeException if the sentiment analysis API is not working, throw the IOExeption
+   */
+  private double calculateSentiAnalysisMagnitude(Sentiment sentiment) {
```

Comment:
I would remove both the `calcualateSentiAnalysisScore` and  `calculateSentiAnalysisMagnitude` method. Now that most of the actual interaction with the SentimentAnalysis service is being done elsewhere, there is no "calculation" being done in these methods and they instead serve primarily as "getters". 

In addition, some of the null checks are redundant since you'll end up checking the nullability of each `Sentiment` object twice.

I question whether we need these null checks at all, since the javadoc for this service does not indicate that it will return any null responses:
[analyzeSentiment(Document document)](https://googleapis.dev/java/google-cloud-language/latest/com/google/cloud/language/v1/LanguageServiceClient.html#analyzeSentiment-com.google.cloud.language.v1.Document-)